File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS07/OtherRecipientInfo.cs
using System;
using System.Runtime.Serialization;

//	OtherRecipientInfo ::= SEQUENCE {
//		oriType		OBJECT IDENTIFIER,
//		oriValue	ANY DEFINED BY oriType 
//	}

namespace Aladdin.ASN1.ISO.PKCS.PKCS7
{
	[Serializable]
	public class OtherRecipientInfo : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<ObjectIdentifier>().Factory(), Cast.N), 
			new ObjectInfo(    ImplicitCreator				    .Factory  , Cast.N), 
		}; 
		// конструктор при сериализации
        protected OtherRecipientInfo(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public OtherRecipientInfo(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public OtherRecipientInfo(ObjectIdentifier oriType, IEncodable oriValue) : 
			base(info, oriType, oriValue) {}

		public ObjectIdentifier	OriType		{ get { return (ObjectIdentifier)this[0]; } } 
		public IEncodable		OriValue	{ get { return                   this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs7/OtherRecipientInfo.java
package aladdin.asn1.iso.pkcs.pkcs7;
import aladdin.asn1.*; 
import java.io.*; 

//	OtherRecipientInfo ::= SEQUENCE {
//		oriType		OBJECT IDENTIFIER,
//		oriValue	ANY DEFINED BY oriType 
//	}

public final class OtherRecipientInfo extends Sequence<IEncodable>
{
    private static final long serialVersionUID = 2144762522291049859L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(ObjectIdentifier.class).factory(), Cast.N), 
		new ObjectInfo(    ImplicitCreator					    .factory  , Cast.N), 
	}; 
	// конструктор при раскодировании
	public OtherRecipientInfo(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public OtherRecipientInfo(ObjectIdentifier oriType, IEncodable oriValue) 
	{
		super(info, oriType, oriValue); 
	}
	public final ObjectIdentifier	oriType	() { return (ObjectIdentifier)get(0); } 
	public final IEncodable         oriValue() { return                   get(1); }
}
